src/ports/nfts/rentalsNFTComponent.ts

```diff
@@ -0,0 +1,140 @@
+import { Contract, NFTFilters } from '@dcl/schemas'
+import { IRentalsComponent, RentalAsset } from '../rentals/types'
+import { INFTsComponent, NFTResult } from './types'
+
+// The recommended maximum amount of elements a subgraph filter_in can handle.
+// For example, if we had a tokenId_in filter, the maximum amount of tokenIds provided would be this number.
+export const MAX_SUBGRAPH_QUERY_IN_ELEMENTS = 500
+
+/**
+ * This component allows users to retrieve Lands and Estates by owner that are currently owned by the rentals contract.
+ * Once a rental starts, the asset is transferred to the rentals contract, so when we want to use the nft server to retrieve
+ * the assets owned by an address, conventionally, the assets that were rented would not be returned because the owner is the rentals
+ * contract.
+ * This component will first fetch the rentals subgraph to check which assets are currently rented by a given address, and then look them up
+ * in the marketplace subgraph to retrieve all the nft data.
+ * @param options.rentalsComponent - The rentals component which allows querying the rentals subgraph.
+ * @param options.marketplaceNFTsComponent - The marketplace component which allows querying the marketplace subgraph.
+ * @param options.contractAddresses - The addresses of the Land and Estate contracts which will be used to filter the results.
+ */
+export function createRentalsNFTComponent(options: {
+  rentalsComponent: IRentalsComponent
+  marketplaceNFTsComponent: INFTsComponent
+  contractAddresses: ReturnType<typeof getLandAndEstateContractAddresses>
+}): INFTsComponent {
+  const {
+    rentalsComponent,
+    marketplaceNFTsComponent,
+    contractAddresses,
+  } = options
+
+  // Query the rentals subgraph to obtain a list of assets where the provided options.owner is the lessor.
+  async function getRentalAssets(options: NFTFilters): Promise<RentalAsset[]> {
```

Comment:
I'd update this here 
```suggestion
  async function getRentalAssets(options: NFTFilters & { owner: string }): Promise<RentalAsset[]> {
```
so we don't have to use the `!` down below.

Reply:
Doing so breaks things up for the caller, I rather just keep it as it is and have it well documented.